webviews/components/history_view.svelte

```diff
@@ -1,16 +1,175 @@
+          wfts.push(currWFT)
+          currWFT.events.push(ev)
+          currWFT.status = "FAILED"
+          break
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT:
+        // wfts.push(currWFT)
+        // currWFT.events.push(ev)
+        // currWFT.status = "TIMED_OUT"
+        // break
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_TASK_SCHEDULED:
+          currWFT.scheduled = tsToDate(ev.eventTime!)
+          break
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_TASK_STARTED:
+          currWFT.started = tsToDate(ev.eventTime!)
+          currWFT.startedEventId = +ev.eventId!
+          break
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_CANCELED:
+        case temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_STARTED:
+          // TODO: append as a command
+          wfts[wfts.length - 1].events.push(ev)
+          break
+        default:
+          currWFT.events.push(ev)
+          break
+      }
+    }
+    console.log(wfts)
+    return wfts
+  }
+
+  //get labelTextForHistoryEvent
+  function labelTextForHistoryEvent(history: temporal.api.history.v1.IHistoryEvent) {
+    const { eventType } = history
+    if (eventType === undefined || eventType === null) {
+      throw new TypeError("Expected history event `eventType` to be defined")
+    }
+    const eventTypeName: string = temporal.api.enums.v1.EventType[eventType]
+      .replace(/^EVENT_TYPE_/, "")
+      .split("_")
+      .map((p) => `${p[0]}${p.substr(1).toLowerCase()}`)
+      .join("")
+    currentEvent = eventType
+    console.log(currentEvent)
+    switch (eventType) {
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED:
+        startTime = tsToDate(history.eventTime).getTime() / 1000
+        return ` ⇨ ${eventTypeName} (${history.workflowExecutionStartedEventAttributes?.workflowType?.name})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        console.log(tsToDate(history.eventTime).getTime() / 1000, "-", startTime, "=", duration)
+        return ""
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
```

Comment:
Please separate calculating the duration from this function.
Create a separate function for calculating it and take the last event as input.